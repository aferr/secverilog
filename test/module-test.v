`include "deptype-example.v"

module testmodule(
  input {L} clk,
  output reg [15:0] {L} out		 
  );   

   wire [15:0] 	    {H} data1 = 0;     
   wire [15:0] 	    {H} data2 = 100;
   reg [15:0] 	    {L} timer;
 
   wire [15:0] 	    {L} out1;
   wire [15:0] 	    {L} out2;
 
  deptype test1
    (
     .clk (clk),
     .timer (timer),
     .data (data1), //should violate an assumption
     .out (out1)
     );

  deptype test2
    (
     .clk (clk),
     .timer (timer),
     .data (data2), //does not violate the assumption
     .out (out2)
     );

   always@(posedge clk)
     begin
	timer <= timer + 1;	
     end

   always@(posedge clk)
     begin
	out <= (timer[0] == 0) ? out1 : out2;
     end
endmodule
