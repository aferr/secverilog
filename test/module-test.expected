Start base conditions are satisfiable? (should be sat)
sat
End base conditions are satisfiable?
unsat
unsat
sat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
unsat
